File: src/main/java/com/cesello/memento/Caretaker.java
package com.cesello.memento;

import java.util.Deque;
import java.util.LinkedList;

public class Caretaker {

    private Deque<EmployeeMemento> employeeHistory = new LinkedList<>();

    public void save(Employee employee) {
        employeeHistory.push(employee.save());
    }

    public void revert(Employee employee) {
        employee.revert(employeeHistory.pop());
    }
}

File: src/main/kotlin/com/cesello/memento/Caretaker.kt
package com.cesello.memento

import java.util.Deque
import java.util.LinkedList

class Caretaker(private val employeeHistory: Deque<EmployeeMemento> = LinkedList()) {

    fun save(employee: Employee) = employeeHistory.push(employee.save())

    fun revert(employee: Employee) = employee.revert(employeeHistory.pop())
}